lib/core/utils/valid-langs.js

```diff
@@ -61,45 +6,12 @@ const langs = [,[,[1,1,1,1,1,1,1,1,1,1,,1,1,1,1,1,1,1,,1,1,1,1,1,1,,1],[1,1,1,1,
  * @returns {Boolean}
  */
 function isValidLang(lang) {
-  let array = langs;
-
-  // padEnd is not supported in IE11
-  while (lang.length < 3) {
-    lang += '`';
-  }
-  for (let i = 0; i <= lang.length - 1; i++) {
-    const index = lang.charCodeAt(i) - 96;
-    array = array[index];
-    if (!array) {
-      return false;
-    }
+  try {
+    // @ts-ignore getCanonicalLocales not in draft but, available on all main browsers
```

Comment:
This sets up the project for future typescript changes. A lot of unused modules in the library adding some weight that ts could eliminate during development.